### PR TITLE
Added type hierarchy check for @WebServlet diagnostic (#159)

### PR DESCRIPTION
**Pull request by tiganov**
**DRAFT:** The tests for @WebServlet diagnostics fail because my fix will report a diagnostic that the WebServlet class does not extend `HttpServlet` when in fact it does. This is because under the test environment `typeHierarchy.getAllSupertypes(type);` is empty. I've tried the test code in my eclipse instance with my change and no diagnostic is reported.

### CODE REVIEW

**Comment by kathrynkodama** [comment]
Looking at the test failures:
```
expected:<[Diagnostic [
  range = Range [
    start = Position [
      line = 9
      character = 0
    ]
    end = Position [
      line = 9
      character = 13
    ]
  ]
  severity = Error
  code = Either [
    left = CompleteHttpServletAttributes
    right = null
  ]
  codeDescription = null
  source = "jakarta-servlet"
  message = "The 'urlPatterns' attribute or the 'value' attribute of the WebServlet annotation MUST be specified."
  tags = null
  relatedInformation = null
  data = null
]]> but was:<[Diagnostic [
  range = Range [
    start = Position [
      line = 10
      character = 13
    ]
    end = Position [
      line = 10
      character = 30
    ]
  ]
  severity = Error
  code = Either [
    left = ExtendHttpServlet
    right = null
  ]
  codeDescription = null
  source = "jakarta-servlet"
  message = "Classes annotated with @WebServlet must extend the HttpServlet class."
  tags = null
  relatedInformation = null
  data = null
], Diagnostic [
  range = Range [
    start = Position [
      line = 9
      character = 0
    ]
    end = Position [
      line = 9
      character = 13
    ]
  ]
  severity = Error
  code = Either [
    left = CompleteHttpServletAttributes
    right = null
  ]
  codeDescription = null
  source = "jakarta-servlet"
  message = "The 'urlPatterns' attribute or the 'value' attribute of the WebServlet annotation MUST be specified."
  tags = null
  relatedInformation = null
  data = null
]]>
	at org.eclipse.lsp4jakarta.jdt.servlet.JakartaServletTest.CompleteWebServletAnnotation(JakartaServletTest.java:73)

```

The [CompleteWebServletTest](https://github.com/eclipse/lsp4jakarta/blob/main/jakarta.jdt/org.eclipse.lsp4jakarta.jdt.test/src/main/java/org/eclipse/lsp4jakarta/jdt/servlet/JakartaServletTest.java#L60) is failing as it detects two diagnostics but only expects 1. 

With your new solution, the `extends HttpServlet` diagnostic is being detected if it exists anywhere in the project, rather than just the open file. Since the files are in the same project: https://github.com/eclipse/lsp4jakarta/tree/main/jakarta.jdt/org.eclipse.lsp4jakarta.jdt.test/projects/jakarta-sample/src/main/java/io/openliberty/sample/jakarta/servlet it is picking up the `extends HttpServlet` even though the problem does not exist in the currently open file.

This seems like an improvement and then we should adjust the `CompleteWebServletTest` to expect both diagnostics (or combine the two test cases)

**Comment by tiganov** [comment]
For some reason `getAllSupertypes()` returns empty when it should return `HttpServlet` for `InvalidWebServlet`, but only when running that test. If I take the test file and run it in an Eclipse instance, no diagnostic is reported because `getAllSupertypes()` returns the correct values. I've disabled the tests as per @kathrynkodama's instructions. Once this PR goes through, I'll open a new issue for fixing this test problem.

File: jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/core/TypeHierarchyUtils.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
+package org.eclipse.lsp4jakarta.jdt.core;
```
Comment (kathrynkodama) [inline jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/core/TypeHierarchyUtils.java:1]: This class needs a header, see an example: https://github.com/eclipse/lsp4jakarta/blob/main/jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/core/jax_rs/NoResourcePublicConstructorQuickFix.java#L1-L12

File: jakarta.jdt/org.eclipse.lsp4jakarta.jdt.test/src/main/java/org/eclipse/lsp4jakarta/jdt/servlet/JakartaServletTest.java [hunk 2]
```diff
@@ -57,6 +59,7 @@ public void ExtendWebServlet() throws Exception {
     }
 
     @Test
+    @Ignore
```
Comment (kathrynkodama) [inline jakarta.jdt/org.eclipse.lsp4jakarta.jdt.test/src/main/java/org/eclipse/lsp4jakarta/jdt/servlet/JakartaServletTest.java:62]: Let's add a comment indicating why we are ignoring this test for now

**Review by kathrynkodama — APPROVED** [review]

**Pull request merged**